src/main/scala/goatrodeo/util/FileWalker.scala

```diff
@@ -263,296 +201,210 @@ object FileWalker {
-            }
+    val fis = in()
 
+    try {
+      val input: ArchiveInputStream[ArchiveEntry] = factory
+        .createArchiveInputStream(
+          fis
+        )
+      val theIterator = Helpers
+        .iteratorFor(input)
+        .filter(!_.isDirectory())
+        .map(ae =>
+          () => {
+            val name = ae.getName()
+
+            (name, input, false)
           }
-
-          val theIterator = compressedIter.filter(!_()._2.isDirectory())
-          Some(theIterator -> (() => input.close()))
-        } else None
-      } catch {
-        case e: Throwable => fis.close(); None
-      }
-    } match {
-      case Success(Some(iter)) =>
-        Some(iter)
-      case Success(None) =>
-        logger.trace("Success(None) ??")
-        None
-      case Failure(e) =>
-        logger.trace(
-          s"Failed to get an Apache Commons Compressed Wrapper: ${e.getMessage}"
         )
-        None
+      Some(theIterator, "Apache Common Wrapper", (() => input.close()))
+    } catch {
+      case e: Exception =>
+        fis.close(); None
     }
   }
 
-  // /**
-  //  * Process Ruby Gem dependency files. These archives are suffixed `.gem`, but are tarballs with 3 files:
-  //  * - metadata.gz - a gzipped file containing the metadata (dependency info, etc) and Gem Spec
-  //  * - checksums.yaml.gz - a gzipped YAML file with the checksums for the archive
-  //  * - data.tar.gz - a tarball containing the actual ruby dependency code
-  //  *
-  //  * This is a separate method from `asApacheCommonsArchiveWrapper`, to allow us to do anything extra
-  //  * with checksums or metadata or such
-  //  * @param in the file to try to construct the stream from
-  //  * @return OptionalArchiveStream
-  //  * */
-  // private def asGemWrapper(in: ArtifactWrapper): OptionalArchiveStream = {
-  //   logger.trace(s"asGemWrapper($in)")
-  //   val factory = (new ArchiveStreamFactory())
-  //   Try {
-
-  //     {
-  //       val fis = in.asStream()
-
-  //       try {
-  //         val input: ArchiveInputStream[ArchiveEntry] = factory
-  //           .createArchiveInputStream(
-  //             fis
-  //           )
-  //         val theIterator = Helpers
-  //           .iteratorFor(input)
-  //           .filter(!_.isDirectory())
-  //           .map(ae =>
-  //             () => {
-  //               val name = ae.getName()
-
-  //               val wrapper = buildWrapper(name, ae.getSize(), () => input)
-
-  //               (name, wrapper)
-  //             }
-  //           )
-  //         Some(theIterator -> (() => input.close()))
-  //       } catch {
-  //         case e: Throwable => fis.close(); None
-  //       }
-  //     }
-  //   }.toOption.flatten
-  // }
+  val definitelyNotArchive = Set(
+    "application/java-vm",
+    "text/plain",
+    "multipart/appledouble",
+    "application/xml-dtd",
+    "application/xml",
+    "text/x-java-properties",
+    "text/x-scala",
+    "text/x-java-source",
+    "application/x-sh",
```

Comment:
I regret to inform you [shar.zip](https://github.com/user-attachments/files/18645124/shar.zip)

Reply:
Sorry... what is the issue with shar.zip ? Missing what I need to enhance

Reply:
Oh, sorry. It's not like, super critical, but shell files can be archives (using sharutils), and there's one in that zip file. I fully expect we'd not support it but just archive formats are wild